src/models/aspera-sdk.model.ts

```diff
@@ -119,13 +121,46 @@ export class ActivityTracking {
       return;
     }
 
+    this.lastWebSocketEvent = webSocketEvent;
+
+    this.notifyWebSocketEvent(webSocketEvent);
+  }
+
+  /**
+   * Notify all consumers when a connection webSocketEvent occurs.
+   *
+   * @param webSocketEvent the event type.
+   */
+  private notifyWebSocketEvent(webSocketEvent: WebsocketEvent): void {
+    if (this.lastNotifiedWebSocketEvent === webSocketEvent) {
+      return;
+    }
+
     this.event_callbacks.forEach(callback => {
       if (typeof callback === 'function') {
         callback(webSocketEvent);
       }
     });
 
-    this.lastWebSocketEvent = webSocketEvent;
+    this.lastNotifiedWebSocketEvent = webSocketEvent;
```

Comment:
Maybe move this up before we trigger the callbacks? That way we can ensure that we don't accidentally trigger callbacks twice for the same message. For example, the scenario where we get a duplicate message before this function has had a chance to iterate/trigger all the registered listeners.

Reply:
Yeah, makes sense!